themes/book/app/routes/$.tsx

```diff
@@ -102,10 +102,32 @@ export function ArticlePageAndNavigation({
           {children}
         </article>
       </TabStateProvider>
-    </UiStateProvider>
+    </>
   );
 }
 
+export function ArticlePageAndNavigation({
+  children,
+  hide_toc,
+  projectSlug,
+  inset = 20, // begin text 20px from the top (aligned with menu)
+}: {
+  hide_toc?: boolean;
+  projectSlug?: string;
+  children: React.ReactNode;
+  inset?: number;
+}) {
+  return (
+    <UiStateProvider>
+      <ArticlePageAndNavigationInternal
+        children={children}
+        hide_toc={hide_toc}
+        projectSlug={projectSlug}
+        inset={inset}
+      />
+    </UiStateProvider>
```

Comment:
Is there a nicer way to access the result of a `useContext` _inside_ the component tree?

Reply:
do you mean that you want to access the UIStateProvider context within this same function, rather than creating a new wrapper component? you can do something like this but it's not nicer...

```
return (
    <UiStateProvider>
        {(() => {
            const uiState = useUiState();
            return (
                <ArticlePageAndNavigationInternal
                    children={children}
                    hide_toc={hide_toc}
                    projectSlug={projectSlug}
                    inset={inset}
                />);
            })
        }
    </UiStateProvider>
```

Reply:
Oh yep, that's one way!